ethers-core/src/types/transaction/eip1559.rs

```diff
@@ -283,3 +288,63 @@ impl From<&Transaction> for Eip1559TransactionRequest {
         }
     }
 }
+
+/// Deserialize either the `input` or the `data` field, preferring the former.
+///
+/// See <https://github.com/ethereum/go-ethereum/pull/28078>
+fn deserialize_input_or_data<'d, D: Deserializer<'d>>(d: D) -> Result<Option<Bytes>, D::Error> {
+    #[derive(Deserialize)]
+    struct InputOrData {
+        input: Option<Bytes>,
+        data: Option<Bytes>,
+    }
+
+    let InputOrData { input, data } = InputOrData::deserialize(d)?;
+
+    Ok(input.or(data))
+}
+
+fn serialize_input_or_data<S: Serializer>(data: &Option<Bytes>, s: S) -> Result<S::Ok, S::Error> {
+    #[derive(Serialize)]
+    struct InputOrData<'a> {
+        data: &'a Option<Bytes>,
+    }
+
+    InputOrData { data }.serialize(s)
+}
```

Comment:
this we don't need, right?

Reply:
or is this to make flatten work?

Reply:
Yeah, I thought so too, but unfortunately the `flatten` trick which makes both `input` and `data` available for `deserialize_input_or_data` means that during serialization it also wants to flatten the results, which it can't because it's a `String`. This returns a `{"data": "0x..."}` object which can be flattened :disappointed:

Reply:
Removed it to keep it simple.